tests/lib/nm/linux_bridge_port_vlan_test.py

```diff
@@ -0,0 +1,255 @@
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+import pytest
+
+from unittest import mock
+
+from libnmstate import nm
+from libnmstate.nm.bridge_port_vlan import PortVlanFilter
+from libnmstate.schema import LinuxBridge as LB
+
+
+ACCESS_TAG = 4000
+
+
+@pytest.fixture
+def NM_mock():
+    with mock.patch.object(nm.user.nmclient, "NM") as m:
+        yield m
+
+
+@pytest.mark.parametrize(
+    "trunk_tags",
+    [
+        [],
+        [{LB.Port.Vlan.TrunkTags.ID: 100}],
+        [{LB.Port.Vlan.TrunkTags.ID: 100}, {LB.Port.Vlan.TrunkTags.ID: 101}],
+        [
+            {
+                LB.Port.Vlan.TrunkTags.ID_RANGE: {
+                    LB.Port.Vlan.TrunkTags.MIN_RANGE: 100,
+                    LB.Port.Vlan.TrunkTags.MAX_RANGE: 199,
+                }
+            }
+        ],
+        [
+            {
+                LB.Port.Vlan.TrunkTags.ID_RANGE: {
+                    LB.Port.Vlan.TrunkTags.MIN_RANGE: 100,
+                    LB.Port.Vlan.TrunkTags.MAX_RANGE: 199,
+                }
+            },
+            {
+                LB.Port.Vlan.TrunkTags.ID_RANGE: {
+                    LB.Port.Vlan.TrunkTags.MIN_RANGE: 500,
+                    LB.Port.Vlan.TrunkTags.MAX_RANGE: 1000,
+                }
+            },
+        ],
+        [
+            {LB.Port.Vlan.TrunkTags.ID: 100},
+            {
+                LB.Port.Vlan.TrunkTags.ID_RANGE: {
+                    LB.Port.Vlan.TrunkTags.MIN_RANGE: 500,
+                    LB.Port.Vlan.TrunkTags.MAX_RANGE: 1000,
+                }
+            },
+        ],
+    ],
+    ids=[
+        "no-trunk-tags",
+        "single-trunk-tag",
+        "two-single-trunk-tags",
+        "tag-ranges",
+        "multiple-tag-ranges",
+        "mixed-single-tag-and-tag-ranges",
+    ],
+)
+@pytest.mark.parametrize(
+    "is_native_vlan", [False, True], ids=["not-native-vlan", "native-vlan"]
+)
+def test_generate_nm_vlan_filtering_config(trunk_tags, is_native_vlan):
+    port_vlan_filter = PortVlanFilter()
+    port_vlan_filter.create_configuration(
+        trunk_tags, ACCESS_TAG if is_native_vlan else None, is_native_vlan
+    )
+    port_bridge_vlan_list = port_vlan_filter.to_nm()
+    expected_trunk_tag_length = len(trunk_tags) + (1 if is_native_vlan else 0)
+    assert expected_trunk_tag_length == len(port_bridge_vlan_list)
+    assert is_native_vlan == any(
+        [
+            port_bridge_vlan.is_pvid() and port_bridge_vlan.is_untagged()
+            for port_bridge_vlan in port_bridge_vlan_list
+        ]
+    )
+    desired_trunk_tags = _port_vlan_trunk_tags_to_tag_list(
+        trunk_tags, ACCESS_TAG, is_native_vlan
+    )
+    _assert_port_settings_vlan_conf(port_bridge_vlan_list, desired_trunk_tags)
+
+
+@pytest.mark.parametrize(
+    "trunk_tags",
+    [
+        [],
+        [100],
+        [100, 200],
+        [(100, 200)],
+        [100, (200, 300)],
```

Comment:
Please choose one type and use it all over. Using int, tuple and a mix of them makes it complicated to manage.
You can use a range always and use (x, x) for single tag.